### ISSUE TITLE
reconnect and closed connection callbacks not called in 6.6.1

### ISSUE THREAD

**Issue by clippered** [mosquito/aio-pika#332]
Looks like the callbacks added via `connection.add_reconnect_callback` and `connection.add_close_callback` were not called after disconnection from the broker.

**Comment by spumer** [comment]
Similar issue for Channel https://github.com/mosquito/aio-pika/issues/331

**Comment by bmario** [comment]
Yeah, I just stumbled over this issue myself. The culprit should be [this commit](https://github.com/mosquito/aio-pika/pull/316/commits/5d3d5451ef33d1e31828bb10cef45e037c44b6f8). All callbacks are now stored in `WeakSet`s and thus, whenever they go out of scope, the garbage collector deletes them.
